examples/custom-mods/README.md

```diff
@@ -0,0 +1,349 @@
+
+As described [here](https://flower.ai/docs/framework/how-to-use-built-in-mods.html#what-are-mods), Flower Mods in their simplest form can be described as:
+
+```python
+def basic_mod(msg: Message, context: Context, app: ClientApp) -> Message:
+    # Do something with incoming Message (or Context)
+    # before passing to the inner ``ClientApp``
+    msg = app(msg, context)
+    # Do something with outgoing Message (or Context)
+    # before returning
+    return msg
+```
+
+and used when defining the `ClientApp`:
+
+```python
+app = fl.client.ClientApp(
+    client_fn=client_fn,
+    mods=[basic_mod],
+)
+```
+
+Note that in this specific case, this mod won't modify anything, and perform FL as usual.
+
+### WandB Flower Mod
+
+If we want to write a mod to monitor our client-side training using [Weights & Biases](https://github.com/wandb/wandb), we can follow the steps below.
+
+First, we need to initialize our W&B project with the correct parameters:
+
+```python
+wandb.init(
+    project=...,
+    group=...,
+    name=...,
+    id=...,
+    resume="allow",
+    reinit=True,
+)
+```
+
+In our case, the group should be the `run_id`, specific to a `ServerApp` run, and the `name` should be the `node_id`. This will make it easy to navigate our W&B project, as for each run we will be able to see the computed results as a whole or for each individual client.
+
+The `id` needs to be unique, so it will be a combination of `run_id` and `node_id`.
+
+In the end we have:
+
+```python
+def wandb_mod(msg: Message, context: Context, app: ClientAppCallable) -> Message:
+    run_id = msg.metadata.run_id
+    group_name = f"Workload ID: {run_id}"
+
+    client_id = str(msg.metadata.dst_node_id)
+    run_name = f"Client ID: {client_id}"
+
+    wandb.init(
+        project="Mod Name",
+        group=group_name,
+        name=run_name,
+        id=f"{run_id}{client_id}",
+        resume="allow",
+        reinit=True,
+    )
+```
+
+Now, before the message is processed by the server, we will store the starting time and the round number, in order to compute the time it took the client to perform its fit step.
+
+```python
+round = int(msg.metadata.group_id)
+start_time = time.time()
+```
+
+And then, we can send the message to the client:
+
+```python
+bwd = app(msg, context)
+```
+
+And now, we the message we got back, we can gather our metrics:
+
+```python
+msg_type = bwd.metadata.message_type
+
+if msg_type == MESSAGE_TYPE_FIT:
+
+    time_diff = time.time() - start_time
+
+    metrics = bwd.content.configs_records
+
+    results_to_log = dict(
+        metrics.get(f"{msg_type}res.metrics", ConfigsRecord())
+    )
+
+    results_to_log[f"{msg_type}_time"] = time_diff
+```
+
+Note that we store our metrics in the `results_to_log` variable and that we only initialize this variable when our client is sending back fit results.
+
+Finally, we can send our results to W&B using:
+
+```python
+wandb.log(results_to_log, step=int(round), commit=True)
```

Comment:
We should use `server_round` instead of `round` b/c round overlaps with the Python built-in function `round`